core/src/main/java/com/microsoft/applicationinsights/internal/authentication/AzureMonitorRedirectPolicy.java

```diff
@@ -0,0 +1,71 @@
+package com.microsoft.applicationinsights.internal.authentication;
+
+// This is a copy from Azure Monitor Open Telemetry Exporter SDK AzureMonitorRedirectPolicy
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.net.HttpURLConnection;
+
+public final class AzureMonitorRedirectPolicy implements HttpPipelinePolicy {
+
+    private static final int PERMANENT_REDIRECT_STATUS_CODE = 308;
+    // Based on Stamp specific redirects design doc
+    private static final int MAX_REDIRECT_RETRIES = 10;
+    private final ClientLogger logger = new ClientLogger(AzureMonitorRedirectPolicy.class);
+    private String redirectedEndpointUrl;
```

Comment:
since it's likely accessed from multiple threads
```suggestion
    private volatile String redirectedEndpointUrl;
```